sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -179,6 +179,7 @@ case class Expand(
 trait GroupingAnalytics extends UnaryNode {
   self: Product =>
   def gid: AttributeReference
+  def groupings: Seq[AttributeReference]
```

Comment:
I have a rough idea for the `GROUPING`, can we do the substitution during the `Expand` node?